middleware/ack.go

```diff
@@ -3,15 +3,46 @@ package middleware
 import (
 	"context"
 
-	amqp "github.com/rabbitmq/amqp091-go"
-
 	amqprpc "github.com/0x4b53/amqp-rpc/v3"
+	amqp "github.com/rabbitmq/amqp091-go"
 )
 
+// OnErrFunc is the function that will be called when the middleware get an
+// error from `Ack`. The error and the delivery will be passed.
+type OnErrFunc func(err error, delivery amqp.Delivery)
+
+// AckLogError is a built-in function that will log the error if any is returned
+// from `Ack`.
+//
+//	middleware := AckDelivery(AckLogError(log.Printf))
+func AckLogError(logFn amqprpc.LogFunc) OnErrFunc {
+	return func(err error, delivery amqp.Delivery) {
+		logFn("could not ack delivery (%s): %v\n", delivery.CorrelationId, err)
+	}
+}
+
+// AckSendOnChannel will first log the error and correlation ID and then try to
+// send on the passed channel. If no one is consuming on the passed channel the
+// middleware will not block but instead log a message about missing channel
+// consumers.
+func AckSendOnChannel(logFn amqprpc.LogFunc, ch chan struct{}) OnErrFunc {
```

Comment:
Since the name begins with `Ack` my first impression was that it Acks messages. Perhaps we can come up with another name?